pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -1062,6 +1105,135 @@ func testHTTPsGoTLSCaptureAlreadyRunningContainer(t *testing.T, cfg *config.Conf
+			context: testContext{
+				extras: make(map[string]interface{}),
+			},
+			postTracerSetup: func(t *testing.T, ctx testContext) {
+				clientSuccess := false
+				var wg sync.WaitGroup
+				wg.Add(1)
+				go func() {
+					time.Sleep(5 * time.Second)
+					clientSuccess = prototls.RunClientOpenssl(t, "localhost", "44330", tlsVersion)
+					wg.Done()
+				}()
+				prototls.RunServerOpenssl(t, "44330", "-www")
+				wg.Wait()
+				if !clientSuccess {
+					t.Fatalf("openssl client failed")
+				}
+			},
+			validation: func(t *testing.T, ctx testContext, tr *Tracer) {
+				// Iterate through active connections until we find connection created above
+				require.Eventuallyf(t, func() bool {
+					payload := getConnections(t, tr)
+					for _, c := range payload.Conns {
+						if c.DPort == 44330 && isTLSTag(c.StaticTags) {
+							return true
+						}
+					}
+					return false
+				}, 4*time.Second, time.Second, "couldn't find TLS connection matching: dstport 44330")
+			},
+		})
+	}
+	tests = append(tests,
+		// run this test at the end
+		tlsTest{
+			name: "http_443_should_not_match",
+			context: testContext{
+				extras: make(map[string]interface{}),
+			},
+			preTracerSetup: func(t *testing.T, ctx testContext) {
+				cfg.EnableHTTPMonitoring = true
+				t.Cleanup(func() { cfg.EnableHTTPMonitoring = false })
+			},
+			postTracerSetup: func(t *testing.T, ctx testContext) {
+				const (
+					serverAddr          = "localhost:44330"
+					expectedOccurrences = 1
+				)
+
+				closeServer := testutil.HTTPServer(t, serverAddr, testutil.Options{
+					EnableTLS: false,
+				})
+				t.Cleanup(closeServer)
+
+				resp, err := nethttp.Get(fmt.Sprintf("http://%s/200/request-test-http44330", serverAddr))
+				require.NoError(t, err)
+				resp.Body.Close()
+			},
+			validation: func(t *testing.T, ctx testContext, tr *Tracer) {
+				// Iterate through active connections until we find connection created above
+				require.Eventuallyf(t, func() bool {
+					payload := getConnections(t, tr)
+					for key := range payload.HTTP {
+						if key.Path.Content == "/200/request-test-http44330" {
+							for _, c := range payload.Conns {
+								if c.SPort == key.SrcPort && c.DPort == key.DstPort && !isTLSTag(c.StaticTags) {
+									return true
+								}
+							}
+						}
+					}
+
+					return false
+				}, 4*time.Second, time.Second, "couldn't find http connection matching: %s", "http://localhost:443/200/request-test-http443")
+			},
+		})
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.teardown != nil {
+				t.Cleanup(func() {
+					tt.teardown(t, tt.context)
+				})
+			}
+			if tt.preTracerSetup != nil {
+				tt.preTracerSetup(t, tt.context)
+			}
+			tr := setupTracer(t, cfg)
+			if tr.ebpfTracer.Type() == connection.EBPFFentry {
+				t.Skip("protocol classification not supported for fentry tracer")
+			}
+			tt.postTracerSetup(t, tt.context)
+			tt.validation(t, tt.context, tr)
+		})
+	}
```

Comment:
adjust your code to fit Bryce's change https://github.com/DataDog/datadog-agent/pull/15970